Change watch start behavior for respect last backup time and type

rumanzo opened the pull request:
I make my backups with systemd service and timer using watch functionality.
There was big surprise, that any service (clickhouse-backup) restart by any reason triggers new full backup, that in my case continues ~20 hours and lead to higher backup storage consumption.
I changed this behavior by adding timeout on start and define next backup type full or increment.

Commit:
respect previous backup time

Commit:
change regexp template
add logging
fix a bug

Commit:
fix typo

Slach commented:
thanks for contribution

Slach commented:
could you fix integration tests failures ? https://github.com/Altinity/clickhouse-backup/actions/runs/7355512267/job/20024306085#step:7:2166

Slach commented:
create `./test/integration/.env` from `.env.example`

you need `docker-compose` for properly test running

use 
```bash
RUN_TESTS=TestServerAPI ./test/integration/run.sh
```
for local development check

rumanzo commented:
> could you fix integration tests failures ? https://github.com/Altinity/clickhouse-backup/actions/runs/7355512267/job/20024306085#step:7:2166

I made local tests that passed fine. But I'll check these tests and will fix it

Commit:
Change time.Sleep to ctx select with time.After for support ctx.Done via api
Change test waiting value due new watch logic

rumanzo commented:
I changed waiting next backup time behavior for respect ctx.Done instead of usual sleep.
Notice that with new logic I changed test expect value too, because new backup waiting watch timeout before continue. It means that if we run watch several times in short time, there will be less completed backups due watch timeout defined in config.
TestServerAPI test passed, also I checked server functionality manually too. All fine, even cancelation in watch wait time for start do first backup.

Please review code logic, code style, etc.
